tests/foreman/ui/test_contenthost.py

```diff
@@ -1250,3 +1260,59 @@ def test_syspurpose_mismatched(session, vm_module_streams):
             'details'
         ]
         assert details['system_purpose_status'] == "Mismatched"
+
+
+@tier4
+def test_pagination_multiple_hosts_multiple_pages(session, module_host_template):
+    """Create hosts to fill more than one page, sort on OS, check pagination.
+
+    Search for hosts based on operating system and assert that more than one page
+    is reported to exist and that more than one page can be accessed. Make some
+    additonal aserts to ensure the pagination widget is working as expected.
+
+    To avoid requiring more than 20 fakes hosts to overcome default page setting of 20,
+    this test will set a new per_page default of 4, using URL method rather than the
+    "entries_per_page" setting to avoid impacting other tests.
+
+    :id: e63e4872-5fcf-4468-ab66-63ac4f4f5dac
+
+    :BZ: 1642549
+    """
+    new_per_page_setting = 3
+    host_num = new_per_page_setting + 1
+    host_name = None
+    start_url = f'/content_hosts?page=1&per_page={new_per_page_setting}'
+    # create more than one page of fake hosts. Need two digits in name to ensure sort order.
+    for count in range(host_num):
+        host_name = 'test-' + str('{:0>2}'.format(count))
+        try:
+            make_fake_host(
+                {
+                    'name': host_name,
+                    'organization-id': module_host_template.organization.id,
+                    'architecture-id': module_host_template.architecture.id,
+                    'domain-id': module_host_template.domain.id,
+                    'location-id': module_host_template.location.id,
+                    'medium-id': module_host_template.medium.id,
+                    'operatingsystem-id': module_host_template.operatingsystem.id,
+                    'partition-table-id': module_host_template.ptable.id,
+                }
+            )
+        finally:
+            pass
+    with session(url=start_url):
+        # search for all the hosts by os. This uses pagination to get more than one page.
+        all_fake_hosts_found = session.contenthost.search(
+            f"os = {module_host_template.operatingsystem.name}"
+        )
+        # assert dump of fake hosts found includes the higest numbered host created for this test
+        match = re.search(r'test-{:0>2}'.format(host_num - 1), str(all_fake_hosts_found))
```

Comment:
Since `host_num` is `new_per_page_setting + 1`, I guess `host_num - 1` is just `new_per_page_setting` :) .
Also, see next comment.

Reply:
but it would not be easier for a new component owner to understand, would it? Anyway, adopting your next suggested change so now this is moot point.